src/main/java/com/atlassian/jira/cloud/jenkins/config/JiraCloudSiteConfig2.java

```diff
@@ -0,0 +1,188 @@
+            if (StringUtils.isEmpty(value)) {
+                return FormValidation.error(
+                        "Site name can't be empty. Paste your Jira Cloud site name here.");
+            }
+
+            if (!SiteValidator.isValid(value)) {
+                return FormValidation.error(
+                        "Site name is invalid. Paste a valid site name, e.g. sitename.atlassian.net.");
+            }
+
+            return FormValidation.ok();
+        }
+
+        public FormValidation doCheckWebhookUrl(@QueryParameter final String value) {
+            if (StringUtils.isEmpty(value)) {
+                return FormValidation.error(
+                        "Webhook URL can’t be blank. Paste it from the Jenkins app in Jira.");
+            }
+
+            if (!WebhookUrlValidator.isValid(value)) {
+                return FormValidation.error("Webhook URL is not a valid URL.");
+            }
+
+            if (!WebhookUrlValidator.containsValidQueryParams(value)) {
+                return FormValidation.error(
+                        "Webhook URL needs to contain query parameter 'jenkins_server_uuid'.");
+            }
+            return FormValidation.ok();
+        }
+
+        @SuppressWarnings("unused")
+        public ListBoxModel doFillCredentialsIdItems(@QueryParameter final String credentialsId) {
+            Jenkins instance = Jenkins.get();
+            if (!instance.hasPermission(Jenkins.ADMINISTER)) {
+                return new StandardListBoxModel().includeCurrentValue(credentialsId);
+            }
+
+            return new StandardListBoxModel()
+                    .includeEmptyValue()
+                    .includeMatchingAs(
+                            ACL.SYSTEM,
+                            instance,
+                            StringCredentials.class,
+                            URIRequirementBuilder.fromUri(ATLASSIAN_API_URL).build(),
+                            CredentialsMatchers.always());
+        }
+
+        @RequirePOST
+        @Restricted(DoNotUse.class) // WebOnly
+        @SuppressWarnings("unused")
+        public FormValidation doTestConnection(
+                @QueryParameter final String site,
+                @QueryParameter final String webhookUrl,
+                @QueryParameter final String credentialsId) {
+            Jenkins.get().checkPermission(Jenkins.ADMINISTER);
+
+            final Optional<String> maybeCloudId = cloudIdResolver.getCloudId("https://" + site);
+
+            if (!maybeCloudId.isPresent()) {
+                return FormValidation.error("Failed to resolve Jira Cloud site: " + site);
+            }
+
+            final Optional<String> maybeSecret = secretRetriever.getSecretFor(credentialsId);
+
+            if (!maybeSecret.isPresent()) {
+                return FormValidation.error("Failed to retrieve secret");
+            }
+
+            // TODO (ARC-1135): call a "ping" endpoint of the Jenkins app to test the connection
```

Comment:
Removed the logic that calls out to Jira to check if the OAuth credentials are valid. We need to replace that with a "ping" call to the Jenkins app, later.